graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/strings/FirstNonNull.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.plugins.pipelineprocessor.functions.strings;
+
+import com.google.common.collect.ImmutableList;
+import org.graylog.plugins.pipelineprocessor.EvaluationContext;
+import org.graylog.plugins.pipelineprocessor.ast.functions.AbstractFunction;
+import org.graylog.plugins.pipelineprocessor.ast.functions.FunctionArgs;
+import org.graylog.plugins.pipelineprocessor.ast.functions.FunctionDescriptor;
+import org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor;
+
+import java.util.List;
+import java.util.Objects;
+
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.object;
+
+public class FirstNonNull extends AbstractFunction<String> {
+    public static final String NAME = "firstNonNull";
+
+    private final ParameterDescriptor<Object, Object> valueParam;
```

Comment:
I believe `<List, List>` is the right signature here: we only accept lists anyway, so we should communicate that to the system, so it can generate meaningful error messages.

Right now passing anything that's not a `List` silently returns `""`, which can be confusing.

Reply:
E.g. if the function accepts `ParameterDescriptor<List, List>` as its value, then the error message becomes:

```
org.graylog.plugins.pipelineprocessor.parser.ParseException: Errors:
 Expected type List for argument value but found String in call to function firstNonNull in line 4 pos 21
```